### PR TITLE
Run bridges zombienet tests on CI

### PR DESCRIPTION
**Pull request by svyatonik**
Brridges zombienet tests are non-standard - zombienet currently missing multiple relay chains support (see e.g. https://github.com/paritytech/zombienet/pull/796), so we need to go live with two relay networks, their parachains + custom test runner (which e.g. doesn't shutdown net when its tests are finished and instead waits for both networks tests to complete). So we are stuck with native zombienet provider => this PR is an attempt to gather everything in a single docker container and run tests there ~Draft, because it is far from finishing - what I want now is to see how it works on CI~

### CODE REVIEW

**Comment by svyatonik** [comment]
(removed review request, because it is not ready yet)

File: .gitlab/pipeline/publish.yml [hunk 1]
```diff
@@ -78,6 +78,7 @@ publish-rustdoc:
       --build-arg VCS_REF="${CI_COMMIT_SHA}"
       --build-arg BUILD_DATE="$(date -u '+%Y-%m-%dT%H:%M:%SZ')"
       --build-arg IMAGE_NAME="${IMAGE_NAME}"
+      --build-arg ZOMBIENET_IMAGE="${ZOMBIENET_IMAGE}"
```
Comment (pepoviola) [inline .gitlab/pipeline/publish.yml:81]: I moved the image definition to `.zombienet-ref` in this pr https://github.com/paritytech/polkadot-sdk/pull/2396/files#diff-037ea159eb0a7cb0ac23b851e66bee30fb838ee8d0d99fa331a1ba65283d37f7L33, so here you will not have the env var defined. I think you can extend from that ref but we should check.

```yaml
  extends:
    - .zombienet-refs
```

**Comment by pepoviola** [comment]
Hi @svyatonik, just as heads-up we are finishing the support of `k8s` in the new version of zombienet that will make spawn and test multiple networks more easy. Maybe we can touch base to collect the needs from bridges and check if we can cover your use case.

Thx!

**Comment by svyatonik** [comment]
> Brridges zombienet tests are non-standard - zombienet currently missing multiple relay chains support (see e.g. [paritytech/zombienet#796](https://github.com/paritytech/zombienet/pull/796)), so we need to go live with two relay networks, their parachains + custom test runner (which e.g. doesn't shutdown net when its tests are finished and instead waits for both networks tests to complete). So we are stuck with native zombienet provider => this PR is an attempt to gather everything in a single docker container and run tests there. Draft, because it is far from finishing - what I want now is to see how it works on CI

Sounds cool! Most of our requirements announced [here](https://github.com/paritytech/zombienet/pull/796#issuecomment-1534260291) and [here](https://github.com/paritytech/zombienet/pull/796#issuecomment-1534300905) are still valid :) So just two relay chains with their parachains + staring external process **that need a way to reach the spawned nodes over ws** is enough for the start.

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 2/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4456201

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable 1/3
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4456200

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: cargo-clippy
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4745147

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: check-runtime-migration-asset-hub-westend
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4789572

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: check-runtime-migration-rococo
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4789571

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: check-runtime-migration-bridge-hub-rococo
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4789575

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: check-runtime-migration-collectives-westend
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4789577

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: check-runtime-migration-westend
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4789570

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: build-malus
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4789613

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: build-runtime-collectives
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4789620

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: build-runtime-starters
**Logs**: https://gitlab.parity.io/parity/mirrors/polkadot-sdk/-/jobs/4789623

File: .gitlab/pipeline/zombienet/bridges.yml [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+  artifacts:
+    name: "${CI_JOB_NAME}_${CI_COMMIT_REF_NAME}_zombienet_bridge_tests"
+    when: always
+    expire_in: 2 days
+    paths:
+      - ./zombienet-logs
+  after_script:
+    - mkdir -p ./zombienet-logs
+    # copy logs of tests runner (run-tests.sh)
+    - cp -r /tmp/bridges-zombienet-tests.*/tmp.*/tmp.* ./zombienet-logs/
```
Comment (serban300) [inline .gitlab/pipeline/zombienet/bridges.yml:41]: Nit: Would be nice if this logs could have a more suggestive name

File: .gitlab/pipeline/zombienet/bridges.yml [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    # copy logs of all nodes
+    - cp /tmp/zombie*/logs/* ./zombienet-logs/
+    # copy logs of our polkadot binary wrapper (polkadot-x.sh)
+    - cp /tmp/polkadot.* ./zombienet-logs/
+    # copy logs of our polkadot-parachain binary wrapper (polkadot-parachain-x.sh)
+    - cp /tmp/polkadot-parachain.* ./zombienet-logs/
```
Comment (serban300) [inline .gitlab/pipeline/zombienet/bridges.yml:47]: I don't see any logs matching this naming pattern. What are these wrappers ?

File: bridges/zombienet/run-tests.sh [hunk 4]
```diff
@@ -1,18 +1,49 @@
...
-export POLKADOT_PARACHAIN_BINARY_PATH=$POLKADOT_SDK_FOLDER/target/release/polkadot-parachain
-export POLKADOT_PARACHAIN_BINARY_PATH_FOR_ASSET_HUB_ROCOCO=$POLKADOT_PARACHAIN_BINARY_PATH
-export POLKADOT_PARACHAIN_BINARY_PATH_FOR_ASSET_HUB_WESTEND=$POLKADOT_PARACHAIN_BINARY_PATH
-export ZOMBIENET_BINARY_PATH=~/local_bridge_testing/bin/zombienet-linux
+
+# set pathc to binaries
+if [ "$ZOMBIENET_DOCKER_PATHS" -eq 1 ]; then
+    export POLKADOT_BINARY_PATH=/usr/local/bin/polkadot
+    export POLKADOT_PARACHAIN_BINARY_PATH=/usr/local/bin/polkadot-parachain
+    export POLKADOT_PARACHAIN_BINARY_PATH_FOR_ASSET_HUB_ROCOCO=/usr/local/bin/polkadot-parachain
```
Comment (serban300) [inline bridges/zombienet/run-tests.sh:30]: Nit: These names seem a bit too verbose. I would try to reduce them a bit. For example something like `ASSET_HUB_ROCOCO_BINARY_PATH`

File: cumulus/scripts/bridges_rococo_westend.sh [hunk 5]
```diff
@@ -132,10 +132,10 @@ LANE_ID="00000002"
 XCM_VERSION=3
 
 function init_ro_wnd() {
-    ensure_relayer
+    local relay_path=$(ensure_relayer)
```
Comment (serban300) [inline cumulus/scripts/bridges_rococo_westend.sh:135]: Nit: Should this be named `relayer_path` ? The same question for  the functions below.

File: .gitlab/pipeline/build.yml [hunk 6]
```diff
@@ -382,3 +382,21 @@ build-subkey-linux:
...
+
+# we need some non-binary artifacts in our bridges+zombienet image
+prepare-bridges-zombienet-artifacts:
+  stage: build
+  extends:
+    - .docker-env
+    - .common-refs
+    - .run-immediately
+    - .collect-artifacts
+  before_script:
```
Comment (serban300) [inline .gitlab/pipeline/build.yml:396]: Nit: Should we also create `/artifacts/bridges-polkadot-sdk/cumulus/scripts` here ? Not sure if needed. Just saw that it's missing.

**Review by serban300 — COMMENTED** [review]
Looks good ! Just left some small comments, mostly nits.

File: .gitlab/pipeline/zombienet/bridges.yml (hunk 3 above)
Comment (svyatonik) [reply .gitlab/pipeline/zombienet/bridges.yml]: Right, I've removed them in latest commits, so this isn't relevant anymore

File: .gitlab/pipeline/zombienet/bridges.yml (hunk 2 above)
Comment (svyatonik) [reply .gitlab/pipeline/zombienet/bridges.yml:41]: Yeah, that's a good idea. But now there are just three log files matching this criteria, so not a big deal imo. When I/we will be adding more tests (hope to do that soon) it'll make sense to give meaningful names to those log files. Let's do it later in a separate PR

**Review by serban300 — APPROVED** [review]

**Review by alvicsam — APPROVED** [review]

**Pull request merged**